ext/afform/core/Civi/Api4/Action/Afform/Revert.php

```diff
@@ -54,6 +54,11 @@ protected function doTask($item) {
       \CRM_Afform_AfformScanner::LAYOUT_FILE,
     ];
 
+    // Check permissions to Revert
+    if (\CRM_Core_Permission::check('manage own afform') && (empty($item['created_id']) || $item['created_id'] !== \CRM_Core_Session::getLoggedInContactID())) {
+      throw new \Civi\API\Exception\UnauthorizedException('You do not have permission to manage this afform.');
+    }
```

Comment:
I think all the test failures are due to not checking if `checkPermissions` is set for the api call.